### DISCUSSION TITLE
Integration into Music-Assistant

### DISCUSSION THREAD

**Discussion by feutl** [B5r1oJ0A9G/teufel_raumfeld#47] [Ideas]
Seems fairly good project and gets pushed by Nabu Casa too ;) perhaps it is possible to get integrated and work together to have Raumfeld being added there.

https://github.com/music-assistant/hass-music-assistant
https://www.youtube.com/watch?v=SEH-DxOsywg
https://github.com/music-assistant/hass-music-assistant/issues/371

This would make it more accessible for lots of people, I am sure :)

**Comment by Mattmazing** [comment]
Hi there! First of all, thanks for this nice Integration. I'm doing my first steps setting it up into Music Assistant. I think it is not needed to directly integrate it, as the speakers show up as Home Assistant Media Players. There is only one thing I'm struggeling. It's also written in the documentation, but I don't know how to get it set up correctly. I have one group consisting of two speakers. The Integration gives me that group to be used as media player group. This group works perfect. When I start anything to play on it, the speakers turn on and play in sync. I only have the problem to do the same on the single speakers. In the docs it's written that the speakers are not foreseen to be used directly, but how can I create a group with just one speaker. I think when I'm getting this done, I can make perfect use of Music Assistant. As the Raumfeld App search often is not working good, it would be a good solution, also to use other streaming sources like Deezer... Many thanks

**Reply by mathebox** [reply]
Having the integration working nicely with Music Assistant would indeed be great. Thanks for giving this a try on your own!
Unfortunately, I'm running HA on hardware that does not support Musik Assistant. So anything from here on is a guess.

> The Integration gives me that group to be used as media player group. 

I use the integration with two Teufel speakers, which I only occasionally group together. In my case, the integration created three HA media players (a grouped one and one for each of the two rooms). Is that also the case for you? If not, consider removing and re-adding the integration in HA _with all speakers turned on_. Maybe that will do the trick.

> In the docs it's written that the speakers are not foreseen to be used directly

Which docs are you referring to?

**Comment by Mattmazing** [comment]
Hi mathebox, 

a full integration in Music Assistant would truely be great. I have 5 speakers of different generations. A new and an old speaker are always grouped together.  
The behaviour of the integration is the same as you discribe it.

The docs I was refering to is the Wiki of this integration.

https://github.com/B5r1oJ0A9G/teufel_raumfeld/wiki

I think I got it wrong, it's written that a single speaker will show up as a group with one speaker. However using a group with two speakers works in MA, using one with one speaker is not working. 

In the end I found that using the provided UpNp/DLNA speakers works better with MA, than using the integration groups. There is also a DLNA group of the grouped speakers available and it plays in sync.

The button support on the devices, and showing pictures within the Raumfeld App is not working, but it's possible to get the text information what ist playing at music assistant on the Raumfeld system.

With the configuration I have found now, i have the possibility of having some extra music providers like Audible and Deezer. It's not perfect but a good possibility to extend the Raumfeld setup. 

Music Assistant is developing fast, maybe I find a solution how to integrate the button support via this integration.

If some pro programmer want to mix this integration directly into MA, it would be nice! I have the feeling that Teufel might stop maintaining the Raumfeld devices in a not so far future. The last update removed Napster from the supported music providers...
